prime-dotnet-webapi/Services/VerifiableCredentialService.cs

```diff
@@ -129,9 +129,19 @@ public async Task<bool> RevokeCredentialsAsync(int enrolleeId)
 
             foreach (var credential in enrolleeCredentials)
             {
-                if (await _verifiableCredentialClient.RevokeCredentialAsync(credential))
+                if (credential.AcceptedCredentialDate == null)
                 {
-                    credential.RevokedCredentialDate = DateTimeOffset.Now;
+                    if (await _verifiableCredentialClient.DeleteCredentialAsync(credential))
+                    {
+                        credential.RevokedCredentialDate = DateTimeOffset.Now;
+                    }
+                }
+                else
+                {
+                    if (await _verifiableCredentialClient.RevokeCredentialAsync(credential))
+                    {
+                        credential.RevokedCredentialDate = DateTimeOffset.Now;
+                    }
```

Comment:
This double conditional can be simplified with either 
`if (credential.AcceptedCredentialDate == null
     ? await _verifiableCredentialClient.DeleteCredentialAsync(credential)
     : await _verifiableCredentialClient.RevokeCredentialAsync(credential))
{
      credential.RevokedCredentialDate = DateTimeOffset.Now;
}`

or

`var success = credential.AcceptedCredentialDate == null
     ? await _verifiableCredentialClient.DeleteCredentialAsync(credential)
     : await _verifiableCredentialClient.RevokeCredentialAsync(credential);

if (success)
{
      credential.RevokedCredentialDate = DateTimeOffset.Now;
}`